src/pages/RFOX/components/Claim/Claim.tsx

```diff
@@ -45,89 +41,75 @@ const ClaimStatus = makeSuspenseful(
   defaultBoxSpinnerStyle,
 )
 
-const ClaimEntries = [ClaimRoutePaths.Select, ClaimRoutePaths.Confirm, ClaimRoutePaths.Status]
-
 export const Claim: React.FC<ClaimRouteProps> = ({ headerComponent, setStepIndex }) => {
-  return (
-    <MemoryRouter initialEntries={ClaimEntries} initialIndex={0}>
-      <ClaimRoutes headerComponent={headerComponent} setStepIndex={setStepIndex} />
-    </MemoryRouter>
-  )
+  return <ClaimRoutes headerComponent={headerComponent} setStepIndex={setStepIndex} />
 }
 
 export const ClaimRoutes: React.FC<ClaimRouteProps> = ({ headerComponent, setStepIndex }) => {
   const location = useLocation()
   const queryClient = useQueryClient()
 
-  const [confirmedQuote, setConfirmedQuote] = useState<RfoxClaimQuote | undefined>()
   const [claimTxid, setClaimTxid] = useState<string | undefined>()
 
-  const stakingAssetAccountAddress = useMemo(() => {
-    return confirmedQuote ? fromAccountId(confirmedQuote.stakingAssetAccountId).account : undefined
-  }, [confirmedQuote])
-
-  const { queryKey: unstakingRequestQueryKey } = useGetUnstakingRequestsQuery({
-    stakingAssetAccountAddress,
-  })
+  const selectedUnstakingRequest = location.state?.selectedUnstakingRequest as
+    | UnstakingRequest
+    | undefined
 
   const handleTxConfirmed = useCallback(async () => {
+    if (!selectedUnstakingRequest) return
+
     await queryClient.invalidateQueries({
-      queryKey: getUnstakingRequestCountQueryKey({
-        stakingAssetId: confirmedQuote?.stakingAssetId,
-        stakingAssetAccountAddress,
-      }),
+      queryKey: [
+        'getUnstakingRequests',
+        { stakingAssetAccountId: selectedUnstakingRequest.stakingAssetAccountId },
+      ],
```

Comment:
Hook fully refactored - see below. Went from 150 lines of not properly cached query leveraging queries to a single `useQueries()` which does one thing for each iteration: get a list of unstaking requests for a given user address (`AccountId`) and staking contract (`stakingAssetId`) combination.

Since part of the hard-to-graspness was the over-abstraction of react-query bits, decided *not* to go with a util once again here, which mitigates the need for navigating to 10s of files.

Reply:
Also note that we go from two invalidations (count and queries) to one, that's actually correct, since now it's as easy as a `useQueries()` of `getUnstakingRequests` queries. 
We don't have an explicit query for count anymore, that's part of `getUnstakingRequests` - making this implementation detail explicit and doing a query of queries made the logic very spaghetti and eventually the caching not work properly, meaning we were always refetching everything when e.g switching accounts in RFOX.